### ISSUE TITLE
Need to be able to choose different forecast options

### ISSUE THREAD

**Issue by inf3rno** [inf3rno/aurora-watch#7]

**Comment by inf3rno** [comment]
In theory our data source: https://www.swpc.noaa.gov/products/aurora-30-minute-forecast has different forecast times. This one is 30 mins, but as far as I remember there is a 1 or a 3 day one too. Afaik. it takes 3 days for the sun flares to reach us, so that is the theoretical maximum currently. I need to check what we have so we can have different options.

**Comment by inf3rno** [comment]
I am not sure how to represent those options or how to use settings to choose between them. For example if we have aurora currently and the 30min forecast shows that we will still have it in 30mins, then I guess I need to start the alert. We need an algorithm which decides when to start the alert, but we need to think it through first. For example if a forecast tells us that we might have aurora in 3 days, then we can send a notification that we might have it in 3 days. After that we can make it more accurate with the 1 day forecast, 1 hour forecast and so on... I am not sure how to do that. I mean for example if the 3 day forecast tells that we will have it, but the 1 days tells that we won't have it, then what to do? Say sorry? The same is true for any pair of times. :S

**Comment by maretree** [comment]
Yes, those times are what I know of only to be. Minimum up to 30min current/present forecast. Then the three day one. 

The thing is... it would be nice to have a little bit of warning for the 30 min alerts... but I don't think that is possible since it can literally change from a 5kp to a 2kp in 5 minutes. =S I think the sooner we can have the warning that it will be a high kp according to your gps coordinates the better. Maybe potential warning of 15 minutes that it will be at your present kp level.

1 and 3 day will be good too because then you will be ready for those last minute kp bursts.

I think currently how it works for the 1-3 day forecast there is no warnings its just something you can read about. The actually apps that give warning are only for the up to 30min in advance. The 1-3 day kp advisory forecasts are just that... potential forecast that that is what is expected. But it's nothing that is set in stone because it's so fluid and even though there may or may not be a projected high kp it really just comes down to what is going on irt. Even sometimes I'm sure that the projected forecast is a normal meh not very promising high kp then all of a sudden out of no where it can be proper storm level.

**Comment by inf3rno** [comment]
Still not clear when to send the notification and if we don't send it how to display the forecast data. Just write a list for different times, something like 3d: 3kp, 1d: 3kp, 30min: 5kp, present: 5kp? Actually I don't like those kp numbers, I'd rather use a color for example green and it would be lighter if the kp is higher and totally dark by 0kp. But still that does not answer the question. Should we display all forecast data, or just the present kp? Or should we display all data and send only notification when the present kp is high or the 30min kp is high?

**Comment by maretree** [comment]
I like the color scale idea! 

Although the numbers are nice too because it can make a difference if I should get out of bed or not. For example how would I know by the color of green that I should get out of bed. I believe my kp is 5.5 so I’m right between the two main kp levels. I don't know that at 2am I want to be dealing with color swatches. Although… I guess I would have only been alerted if it was at the kp level where my gps is detected… So then ok… the color scheme is cool as long as its based upon the kp numbers behind the scene. 

Maybe using the color scheme like how it is on the noaa webpage. 

Light green for lower kp, and bright red for storm level. This would make it consistent with what is already out there. 

I'll have more of a think on the when and list of different times.

**Comment by inf3rno** [comment]
We could use 5 icons for the 5 kp levels instead of a color scale. I thought that because weather forecast sites use icons for cloud and rain too and we can have icon for fullmoon, etc. too.

**Comment by maretree** [comment]
Actually there are 4. 

3, 5, 7, 9

Noaa delineates them as colors. Royal blue, green, yellow, red respectively. 

https://softservenews-6eg3vi45jxatmx1gteg.stackpathdns.com/globeNE_big2.gif <https://softservenews-6eg3vi45jxatmx1gteg.stackpathdns.com/globeNE_big2.gif>
